chromadb/db/sqlite.py

```diff
@@ -0,0 +1,708 @@
+        collection_uuid=None,
+    ) -> Tuple[List[List[uuid.UUID]], npt.NDArray]:
+
+        # Either the collection name or the collection uuid must be provided
+        if collection_name is None and collection_uuid is None:
+            raise TypeError("Arguments collection_name and collection_uuid cannot both be None")
+
+        if collection_name is not None:
+            collection_uuid = self.get_collection_uuid_from_name(collection_name)
+
+        if len(where) != 0 or len(where_document) != 0:
+            results = self.get(
+                collection_uuid=collection_uuid,
+                where=where,
+                where_document=where_document
+            )
+
+            if len(results) > 0:
+                ids = [x[1] for x in results]
+            else:
+                raise NoDatapointsException(
+                    f"No datapoints found for the supplied filter {json.dumps(where)}"
+                )
+        else:
+            ids = None
+
+        index = self._index(collection_uuid)
+        uuids, distances = index.get_nearest_neighbors(embeddings, n_results, ids)
+
+        return uuids, distances
+
+    def create_index(self, collection_uuid: str):
+        """Create an index for a collection_uuid and optionally scoped to a dataset.
+        Args:
+            collection_uuid (str): The collection_uuid to create an index for
+            dataset (str, optional): The dataset to scope the index to. Defaults to None.
+        Returns:
+            None
+        """
+        get = self.get(collection_uuid=collection_uuid)
+
+        uuids = [x[1] for x in get]
+        embeddings = [x[2] for x in get]
+
+        index = self._index(collection_uuid)
+        index.add(uuids, embeddings)
+
+    def add_incremental(self, collection_uuid, uuids, embeddings):
+        index = self._index(collection_uuid)
+        index.add(uuids, embeddings)
+
+    def reset_indexes(self):
+        delete_all_indexes(self._settings)
+        self.index_cache = {}
+
+    def reset(self):
+        self._conn.execute("DROP TABLE collections")
+        self._conn.execute("DROP TABLE embeddings")
+        self._create_table_collections()
+        self._create_table_embeddings()
+        self.commit()
+        self.reset_indexes()
+
+    def __del__(self):
+        logger.info("Exiting: Cleaning up .chroma directory")
+        self.reset_indexes()
+
+    def persist(self):
+        raise NotImplementedError(
+            "Set chroma_db_impl='sqlite+persist' to get persistence functionality"
+        )
+
+
+class PersistentSQLite(SQLite):
+    _save_folder = None
+
+    def __init__(self, settings):
+        super().__init__(settings=settings)
+
+        if settings.persist_directory == ".chroma":
+            raise ValueError(
+                "You cannot use chroma's cache directory .chroma/, please set a different directory"
+            )
+
+        self._save_folder = settings.persist_directory
+        self.load()
```

Comment:
It would be really great if we have it as an option whether to load SQLite db into memory or not. We have seen that SQLite on SSD is really superfast